internal/provider/access_key_resource.go

```diff
@@ -0,0 +1,124 @@
+	return PROVIDER_TYPE_NAME + "_access_key"
+}
+
+func (r *AccessKeyResource) NodeType() string {
+	return "access_key"
+}
+
+func (r *AccessKeyResource) TerraformSchema() schema.Schema {
+	return AccessKeyResourceSchema()
+}
+
+func (r *AccessKeyResource) NotConvertible() bool {
+	// There is no import for access keys, so implement the "not convertible" interface
+	return true
+}
+
+// Configure implements resource.ResourceWithConfigure.
+func (r *AccessKeyResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected client.Client, got: %T. Please report this issue to Mezmo.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// POST access key
+func (r *AccessKeyResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan AccessKeyResourceModel
+	if diags := req.Plan.Get(ctx, &plan); setDiagnosticsHasError(diags, &resp.Diagnostics) {
+		return
+	}
+	access_key := AccessKeyFromModel(&plan)
+
+	stored, err := r.client.CreateAccessKey(access_key)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating access key",
+			"Could not create access key, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	AccessKeyToModel(&plan, stored)
+	diags := resp.State.Set(ctx, plan)
+	setDiagnosticsHasError(diags, &resp.Diagnostics)
+}
+
+// Delete implements resource.Resource.
+func (r *AccessKeyResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state AccessKeyResourceModel
+	if diags := req.State.Get(ctx, &state); setDiagnosticsHasError(diags, &resp.Diagnostics) {
+		return
+	}
+
+	access_key := AccessKeyFromModel(&state)
+
+	err := r.client.DeleteAccessKey(access_key)
+	if client.IsNotFoundError(err) {
+		// If the key wasn't found, just ignore and let TF clean up state on its own
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Deleting Access Key",
+			"Could not delete access key, unexpected error: "+err.Error(),
+		)
+		return
+	}
+}
+
+func (r *AccessKeyResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = r.TypeName()
+}
+
+func (r *AccessKeyResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// There is no getter for access keys. Implement a noop to satisfy the interface.
+}
+
+func (*AccessKeyResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = AccessKeyResourceSchema()
+}
+
+func (r *AccessKeyResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Updating keys is not supported. Noop this to satisfy the interface.
+}
```

Comment:
Not having Update (and Read?) will break if the user updates the key title or something else. Adding this acceptance test highlight the issue.


```go
			{
				Config: GetCachedConfig(cacheKey) + `
					output "http_key" {
						value = mezmo_access_key.for_http.key
						sensitive = true
						description = "the output of key"
					}
					resource "mezmo_access_key" "for_http" {
						title = "updated"
						source_id = mezmo_http_source.my_source.id
					}`,
				Check: resource.ComposeTestCheckFunc(
					resource.TestCheckResourceAttr("mezmo_access_key.for_http", "title", "updated"),
				),
			},
```

```bash
--- FAIL: TestAccessKeyResource (2.57s)
    utils.go:106: Created account for testing 
    access_key_resource_test.go:13: Step 3/3 error: Error running apply: exit status 1
        
        Error: Provider produced inconsistent result after apply
        
        When applying changes to mezmo_access_key.for_http, provider
        "provider[\"registry.terraform.io/hashicorp/mezmo\"]" produced an unexpected
        new value: .title: was cty.StringVal("updated"), but now cty.StringVal("http
        ingestion key").
        
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
```

If we can not add get or update methods to the backend we should fail the update function with a helpful message:

```go
func (r *AccessKeyResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
	resp.Diagnostics.AddError(fmt.Sprintf("Can not update %s after creation", r.TypeName()), "")
}
```

Reply:
There is no update currently allowed for the keys.  For now, let's add the error until we can decide whether or not to support a PATCH for these.

Reply:
There is no GET either, and that's by design.

Reply:
If we error at update an empty get method should be fine I think.